dvc/repo/plots/__init__.py

```diff
@@ -33,3 +104,113 @@ def modify(self, path, props=None, unset=None):
 
         dvcfile = Dvcfile(self.repo, out.stage.path)
         dvcfile.dump(out.stage, update_pipeline=True)
```

Comment:
@Suor, `plots modify` should not generate lockfile at all. Please pass `no_lock=True` here.

Reply:
I'll do it, as I am working on the plots as well.

Reply:
This line is not changed in this PR) Should be fixed separately though.

Reply:
Looks like `.dump_lock()` should be separated from `.dump()` otherwise this error will introduce itself somewhere else.

Reply:
I agree. But, for single stage file, we would have to make it a no-op.

Reply:
That's ok.

Reply:
I think, it's the same thing compared to passing args. We just need to have a sane default.

Reply:
Initially, the idea was not to touch `dvc.yaml` at all, but as we are doing it quite often, it's better to protect `dvc.lock` rather than `dvc.yaml` by default.
So, we could change it to the following default:
```py
def dump(update_pipeline=True, lock=False)
```

The only place were we need to update `lock` are `dvc repro` and `dvc run`, at least atm.